Create asked classes and methods.

NaKraVon opened the pull request:
Created abstract parent class Machine, created Truck, Excavator and Bulldozer child classes and make a doWork and stopWork methods for them.

mr-Edoc committed:
Created abstract parent class Machine, created Truck, Excavator and Bulldozer child classes and make a doWork and stopWork methods for them.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+public class Bulldozer extends Machine {
+    @Override
+    public void doWork() {
+        System.out.print("Bulldozer start working" + "\n");
```
doWork() prints: `"Bulldozer start working"`. The task description requires: "In those classes override `doWork()`, so it will print a message that a certain machine started its work." Consider changing the message to: `"Bulldozer started working"` so it is explicit and matches the requirement. Also follow the checklist: "Write informative messages in methods."

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void stopWork() {
+        System.out.print("Bulldozer start working" + "\n");
```
stopWork() currently prints the start message again (`"Bulldozer start working"`). This violates the description: "Override `stopWork()` as well. It should print messages that certain machines stopped working." Please change to `"Bulldozer stopped working"`.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
```
There is an empty line here. The checklist says: "Please don't add redundant empty lines to your code." Remove this blank line to match the project's formatting rules.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+    public void doWork() {
+        System.out.print("Bulldozer start working" + "\n");
+    }
+
```
There is an extra blank line between method definitions. The checklist requires avoiding redundant empty lines: "Please don't add redundant empty lines to your code." Remove this blank line so methods are placed compactly.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,13 @@
+public class Excavator extends Machine {
+    @Override
+    public void doWork() {
+        System.out.print("Excavator start working" + "\n");
```
**Wrong/ambiguous message in doWork()**
- Current code: `System.out.print("Excavator start working" + "\n");` (line 6).
- Why: The task description requires doWork() to "print a message that a certain machine started its work." Exact text from description: "In those classes override `doWork()`, so it will print a message that a certain machine started its work." 
- Recommendation: Use an explicit past/continuous wording that clearly states the action, for example:
  - `System.out.println("Excavator started working");`
This also satisfies the checklist requirement: "Write informative messages in methods." (Use English and indicate machine type).

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void stopWork() {
+        System.out.print("Excavator start working" + "\n");
```
**Incorrect message in stopWork() — it repeats the start message**
- Current code: `System.out.print("Excavator start working" + "\n");` (line 11).
- Why: This violates the task description requirement: "Override `stopWork()` as well. It should print messages that certain machines stopped working." 
- Recommendation: Change to something like:
  - `System.out.println("Excavator stopped working");`
so the output clearly differentiates starting vs stopping actions and meets the checklist "Write informative messages in methods."

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
```
**Redundant empty line after package declaration**
- Current: there is an empty line at line 2.
- Why: The checklist instructs: "Please don't add redundant empty lines to your code." Remove unnecessary blank lines to match the provided examples.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,13 @@
+    public void doWork() {
+        System.out.print("Excavator start working" + "\n");
+    }
+
```
**Redundant empty line between methods**
- Current: there is an empty line at line 8 between the end of doWork() and the next @Override for stopWork().
- Why: The checklist explicitly asks to avoid redundant empty lines after class declarations or method signatures: "Please don't add redundant empty lines to your code." Keeping method declarations compact improves readability and matches the task guidelines. 
- Recommendation: Remove the blank line so the next `@Override` appears directly after the previous method's closing brace.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
```
There is a redundant empty line between the package declaration and the class declaration. The checklist asks to avoid redundant empty lines: **"Please don't add redundant empty lines to your code."** Remove this blank line so the class follows the package declaration directly.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
```
There is a redundant empty line immediately after the class declaration. The checklist explicitly says: **"We don't need them after class declaration or method signature."** Remove this blank line so the first method follows the class declaration without an extra empty line.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
+    public abstract void doWork();
+
```
There is a redundant empty line after the `doWork()` method signature. According to the checklist: **"We don't need them after class declaration or method signature."** Remove this blank line so `stopWork()` follows directly after `doWork()`.

mateacademy-ai-mentor replied:
There are redundant empty lines inside the file (after the package declaration and between method signatures). This violates the checklist item: "Please don't add redundant empty lines to your code." Remove the extra blank lines so the class is compact (no blank line after the package and no blank lines between the method signatures).

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,17 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] agr) {
```
**Redundant empty line after method signature**
- Checklist violation: "Please don't add redundant empty lines to your code." 
- There is an empty line immediately after the `public static void main(String[] agr) {` signature. Remove that blank line so the method body starts directly on the next line. This matches the checklist's formatting requirement.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,17 @@
+        Machine truck = new Truck();
+        Machine excavator = new Excavator();
+        Machine bulldozer = new Bulldozer();
+
```
**Redundant empty line inside method (remove it)**
- Checklist violation: "Please don't add redundant empty lines to your code." 
- The blank line at this position is unnecessary and should be removed to keep the method body compact and consistent with the provided examples.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,17 @@
+        Machine bulldozer = new Bulldozer();
+
+        Machine[] machines = new Machine[]{truck, excavator, bulldozer};
+
```
**Redundant empty line inside method (remove it)**
- Checklist violation: "Please don't add redundant empty lines to your code." 
- The blank line at this position is unnecessary and should be removed to keep the method body compact and consistent with the provided examples.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
```
Redundant empty line after the package declaration. The checklist says: "Please don't add redundant empty lines to your code." Remove this blank line so the file follows the formatting guidance.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,13 @@
+public class Truck extends Machine {
+    @Override
+    public void doWork() {
+        System.out.print("Truck start working" + "\n");
```
doWork() message is ambiguous/wrong tense. Current: `System.out.print("Truck start working" + "\n");` (line 6). The description requires doWork() to "print a message that a certain machine started its work." Consider changing to: `System.out.println("Truck started working");` to satisfy the checklist: "Write informative messages in methods."

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,13 @@
+    public void doWork() {
+        System.out.print("Truck start working" + "\n");
+    }
+
```
Redundant empty line between methods. The checklist states: "Please don't add redundant empty lines to your code." Remove this blank line so the `@Override` for stopWork() follows directly after the previous method.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void stopWork() {
+        System.out.print("Truck stop working" + "\n");
```
stopWork() message wording does not match the requirement to indicate stopping. Current: `System.out.print("Truck stop working" + "\n");` (line 11). The description requires stopWork() to print that the machine stopped working. Change to: `System.out.println("Truck stopped working");` to comply.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I’m requesting changes because there are critical blockers that violate the task requirements and checklist.

Critical issues to fix:
- Bulldozer.java: `stopWork()` prints a start message instead of a stop message (line 11). Requirement: stopWork() must indicate stopping. Change to something like `System.out.println("Bulldozer stopped working");`. Also adjust `doWork()` to clearly state starting, e.g., `System.out.println("Bulldozer started working");` (line 6).
- Excavator.java: `stopWork()` repeats the start message (line 11). It must say the machine stopped. Use `System.out.println("Excavator stopped working");`. Also make `doWork()` explicit: `System.out.println("Excavator started working");` (line 6).

Minor but required by the checklist (formatting):
- Remove redundant empty lines in all noted files: Bulldozer.java (lines 2, 8), Excavator.java (lines 2, 8), Machine.java (after package, after class decl, and between method signatures), MainApp.java (after main signature and extra blank lines inside), Truck.java (lines 2, 8). The checklist explicitly says: “Please don’t add redundant empty lines to your code.”
- For consistency and clarity, prefer `System.out.println(...)` over `System.out.print(... + "\n")` in all machine messages, and use clear tense: “started working” / “stopped working.” For Truck.java, update messages to `"Truck started working"` and `"Truck stopped working"`.

You’ve done well setting up the class hierarchy and demonstrating polymorphism in MainApp. After correcting the stop messages and tidying the formatting, your solution will fully meet the requirements.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

mr-Edoc committed:
Fixed critical issues:
Change message "start working" in stopWork methods to "stop working".
Fixed minor issues:
1.Remove empty lines in all notes.
2. change print to println.

mr-Edoc committed:
Fixed checkstyle

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+
+public class Bulldozer extends Machine {
+
+    @Override
```
This violates checklist item "Write informative messages in methods." The message should indicate the machine type and whether it is starting or stopping, and should be in clear English. Also remove the explicit `"\n"` concatenation because `System.out.println` already prints a newline. Consider changing to:

`System.out.println("Bulldozer started working");`

instead of the current string concatenation.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Bulldozer start working" + "\n");
+    }
+
```
This violates checklist item "Write informative messages in methods." The stopWork() message should clearly indicate the machine stopped working and should not append `"\n"`. Consider changing to:

`System.out.println("Bulldozer stopped working");`

instead of the current string concatenation.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+
+public class Excavator extends Machine {
+
+    @Override
+    public void doWork() {
+        System.out.println("Excavator start working" + "\n");
+    }
+
```
The printed messages should be informative English phrases that clearly indicate the machine type and whether it started or stopped work. This violates the description checklist items: "In those classes override `doWork()`, so it will print a message that a certain machine started its work." and "Override `stopWork()` as well. It should print messages that certain machines stopped working." Also violates the checklist note: "Write informative messages in methods." Please change the outputs to e.g. `System.out.println("Excavator started working");` and `System.out.println("Excavator stopped working");` and remove the explicit `"\n"` concatenation (System.out.println already adds a newline).

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,15 @@
+        Machine truck = new Truck();
+        Machine excavator = new Excavator();
+        Machine bulldozer = new Bulldozer();
+        Machine[] machines = new Machine[]{truck, excavator, bulldozer};
 
+        for (Machine machine : machines) {
+            machine.doWork();
+            machine.stopWork();
```
There is a redundant empty line inside `main` (between the `Machine[] machines = ...;` statement and the `for` loop). This violates the checklist item: "Please don't add redundant empty lines to your code." Remove the blank line so the method matches the compact example from the checklist. The functionality is correct, only the extra blank line should be removed.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+
+public class Truck extends Machine {
+
+    @Override
```
This violates checklist item "Write informative messages in methods." The doWork() message should explicitly name the machine and indicate it has started working, and you should not append `"\n"` when using `System.out.println`. Consider changing to:

`System.out.println("Truck started working");`

Current line: `System.out.println("Truck start working" + "\n");`

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Truck start working" + "\n");
+    }
+
```
This violates checklist item "Write informative messages in methods." The stopWork() message should explicitly name the machine and indicate it has stopped working, and you should remove the `"\n"` concatenation. Consider changing to:

`System.out.println("Truck stopped working");`

Current line: `System.out.println("Truck stop working" + "\n");`

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

**Decision: REQUEST_CHANGES**

Your solution is very close and the class hierarchy, overrides, and polymorphic usage in `MainApp` look good. However, there are a few checklist violations that must be fixed before approval.

Critical issues to fix (blockers):
- Informative messages and newline handling:
  - `Bulldozer.java`:
    - Line 5: Change to `System.out.println("Bulldozer started working");` (remove `+ "\n"`, use past tense).
    - Line 9: Change to `System.out.println("Bulldozer stopped working");`.
  - `Excavator.java`:
    - Line 5: Change to `System.out.println("Excavator started working");`.
    - Line 9: Change to `System.out.println("Excavator stopped working");`.
  - `Truck.java`:
    - Line 5: Change to `System.out.println("Truck started working");`.
    - Line 9: Change to `System.out.println("Truck stopped working");`.
  - Why: The checklist requires clear English messages identifying the machine and whether it started or stopped. Also, `System.out.println` already adds a newline—do not concatenate `"\n"`.

Minor style fixes (still required by checklist):
- `Machine.java`: Remove redundant blank lines after the package and between method declarations so the class is compact.
- `MainApp.java`: Remove the extra blank line inside `main` (between the `Machine[]` declaration and the `for` loop) and any trailing empty line at the file end.

Once you adjust the messages and tidy the blank lines, the solution will meet all requirements. Nice job on the overall structure and polymorphism; these are just quick, precise edits to align with the checklist.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

mr-Edoc committed:
Fixed checkstyle according to last review

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
```
This violates the checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line directly after the package declaration to keep the file compact.

NaKraVon replied:
Removing this emptyline making Java CL not accept this commit because of their checkstyle policy. All emptyline that can be removed i remove already.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+
```
This violates the checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line immediately after the class declaration so the `@Override` and method start directly under the class opening.

NaKraVon replied:
Removing this emptyline making Java CL not accept this commit because of their checkstyle policy. All emptyline that can be removed i remove already.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Bulldozer started working");
+    }
+
```
This violates the checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line between the two method blocks so the second `@Override` appears directly after the first method's closing brace.

NaKraVon replied:
Removing this emptyline making Java CL not accept this commit because of their checkstyle policy. All emptyline that can be removed i remove already.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
```
This violates the checklist item: **"Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature."** Remove the empty line between the `package` declaration and the `public class Excavator` declaration to keep the file compact.

NaKraVon replied:
Removing this emptyline making Java CL not accept this commit because of their checkstyle policy. All emptyline that can be removed i remove already.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Excavator extends Machine {
+
```
This violates the checklist item: **"Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature."** Remove the empty line immediately after the class declaration so the first method annotation follows the class line without a blank line.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Excavator started working");
+    }
+
```
This violates the checklist item: **"Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature."** Remove the blank line between the end of `doWork()` and the `@Override` for `stopWork()` so the methods appear consecutively.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax;
+
```
This violates checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line at line 2 (the empty line after the package declaration) so the file begins directly with the class declaration.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,7 @@
+
+public abstract class Machine {
+    public abstract void doWork();
+
```
This violates checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line at line 5 (the empty line between the two abstract method declarations) so the two `public abstract` methods appear consecutively.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
```
This violates the checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line directly after the package declaration so the file is more compact.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Truck extends Machine {
+
```
This violates the checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line immediately after the class declaration so the `@Override` and method signature follow the class opening without extra spacing.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,14 @@
+    public void doWork() {
+        System.out.println("Truck started working");
+    }
+
```
This violates the checklist item: "Please don't add redundant empty lines to your code. We don't need them after class declaration or method signature." Remove the blank line between the two method blocks so the second `@Override` appears directly after the first method's closing brace.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Educational approval: Functionality is correct and your OOP design is solid—great job using an abstract `Machine` with concrete `Truck`, `Excavator`, and `Bulldozer` classes, and demonstrating polymorphism in `MainApp` using a `Machine[]`.

What you did well:
- Clear, English messages in `doWork()`/`stopWork()` and proper `public` overrides.
- Correct inheritance and polymorphic iteration in `MainApp`.

Needs improvement (checklist: “Please don't add redundant empty lines…”):
- `Bulldozer.java`: remove the blank line after the `package` (line 2), after the class declaration (line just below `class Bulldozer {`), and the blank line between the two methods. This keeps `@Override` lines directly under the class opening and methods consecutive.
- `Excavator.java`: same fixes—remove the empty line after `package`, after the class declaration, and between `doWork()` and `stopWork()`.
- `Truck.java`: same pattern—delete the blank line after `package`, after the class opening, and between the methods.
- `Machine.java`: remove the blank line after `package` (line 2) and the blank line between the two `public abstract` method declarations so they’re consecutive.
- Optional: In `MainApp.java`, if there’s an extra blank line after `package`, remove it; also consider renaming `agr` to `args` for convention.

Why this matters: removing redundant empty lines improves readability and complies with the project’s style checklist. Quick next step: delete the noted blank lines and re-run checks. Nice work overall—just a small formatting pass and you’re fully compliant!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
